api-put-object-streaming.go

```diff
@@ -348,7 +349,12 @@ func (c Client) putObjectNoChecksum(ctx context.Context, bucketName, objectName
 	}
 	if size > 0 {
 		if isReadAt(reader) && !isObject(reader) {
-			reader = io.NewSectionReader(reader.(io.ReaderAt), 0, size)
+			seeker, _ := reader.(io.Seeker)
+			offset, err := seeker.Seek(0, io.SeekCurrent)
+			if err != nil {
+				return 0, errors.New("error in seeker")
```

Comment:
why not return err as is? The message "error in seeker" does not give actionable message to user of the api.

Reply:
:+1:  - we shouldn't translate errors in any-form to Go native errors, we should in-fact do `ErrInvalidArgument(err.Error())`  such that we have consistent errors being sent.